include/actionspace/ObjectOrientedActionSpace.hpp

```diff
@@ -1,211 +1,203 @@
+        double m_heading_offset;
+
+        friend class ObjectOrientedActionSpace;
+    };
+
+    /// This is an action class that obtains information related to the
+    /// pose of the cube for execution/successor calculation
+    class ObjectOrientedAction : public ActionSpace::Action {
+     public:
+            /// Constructs object oriented action, indicating starting position
+            /// w.r.t the object pose
+            ///
+            /// \param speed : the speed of cozmo, in mm / s, negative speed
+            ///     refers to backward movement
+            /// \param start_pose : starting pose of cozmo's action,
+            ///     its attributes are x(mm), y(mm), and theta(radian)
+        explicit ObjectOrientedAction(
+            const double& speed,
+            const Eigen::Vector3d& start_pose) : \
+            m_speed(speed),
+            m_start_pose(start_pose) {}
+
+        double getSpeed() const { return m_speed; }
+        Eigen::Vector3d getStartPose() const { return m_start_pose; }
+
+     private:
+            double m_speed;
+            Eigen::Vector3d m_start_pose;
+
+            friend class ObjectOrientedActionSpace;
+    };
+
+    /// \param speeds : all possible speeds (mm/s) for the actions
+    ///     (duplicates persist)
+    /// \param ratios : The aspect ratio of sides of the cube, given a
+    ///     rectangular object size(ratios) equals 2. The ratio refers to
+    ///     numeric length (mm) of each unique side length.
+    /// \param edge_offset : The max distance, along the edge of
+    ///     the object, from the center of that edge (mm)
+    /// \param num_offset : number of starting position offsets on each side
+    ///     of the object; this value must always be odd
+    ObjectOrientedActionSpace(
+        const std::vector<double>& speeds,
+        const std::vector<double>& ratios,
+        const double& edge_offset,
+        const int& num_offset = 40);
+
+    ~ObjectOrientedActionSpace() {
+        for (size_t i = 0; i < m_actions.size(); ++i) {
+            delete(m_actions[i]);
+        }
+        m_actions.clear();
+    }
+
+    /// Calculates the similarity between two actions in the action space
+    /// Similarity is defined by the euclidean distance between all
+    /// attributes of an ObjectOrientedAction
+    ///
+    /// \param action_id1, action_id2 : The id's of the actions to compare
+    /// \param[out] similarity : The similarity value between two actions
+    ///     value of 0 means they are identical
+    /// \return true if successful, false otherwise
+    bool action_similarity(
+        const int& action_id1,
+        const int& action_id2,
+        double* similarity) const;
+
+    /// Documentation inherited
+    ActionSpace::Action* get_action(const int& action_id) const;
+
+    /// Documentation inherited
+    bool is_valid_action_id(const int& action_id) const;
+
+    /// Converts a generic action to an object oriented action with respect
```

Comment:
/// Converts a generic action to an object oriented action with respect to the pose of the object.

(Remove everything after this)